lib/output.js

```diff
@@ -597,6 +598,12 @@ function OutputStream(options) {
             return true;
         }
 
+        if (output.option('webkit')
+            && output.parent(0) instanceof AST_PropAccess
+            && output.parent(1) instanceof AST_Assign) {
```

Comment:
```patch
+                && output.parent(1) instanceof AST_Assign) {
```
My issue with this condition is that we really do not know the extent of webkit's bug and it could likely be triggered in other ways with more complex LHS and RHS expressions. That's why I'd recommend dropping this clause altogether for this extremely rare case of reading a property directly from a function expression - it would never occur in the wild anyway.